# Undertanding the cartwheel data set

The notebook aims to undertand the content of the cartwheel data set.


## Acknowledgments

- Data from https://www.coursera.org/ from the course "Understanding and Visualizing Data with Python" by University of Michigan


# Cartwheel data set
Questions and answers. The dataset description
    - The dataset used here is an extension from the original cartwheel dataset from cursera
    - Total numer of observations: 52
    - Many observations/measurements/recordings of the characteristics/attributes/variables of cartwheel executions
    - Variables: Age, Gender, GenderGroup, Glasses, GlassesGroup, Height, Wingspan, CWDistance, ... (X variables)
    


# Importing and inspecting the data

In [27]:
# Define where you are running the code: colab or local
RunInColab          = True     # (False: no  | True: yes)

# If running in colab:
if RunInColab:
    # Mount your google drive in google colab
    from google.colab import drive
    drive.mount('/content/drive')

    # Find location
    #!pwd
    #!ls
    #!ls "/content/drive/My Drive/Colab Notebooks/MachineLearningWithPython/"

    # Define path del proyecto
    Ruta            = "/content/drive/MyDrive/NotebooksProfessor"

else:
    # Define path del proyecto
    Ruta            = ""

Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).


In [34]:
# Import the packages that we will be using
import matplotlib.pyplot as plt
import pandas as pd

# Dataset url
url = Ruta + "/datasets/iris/iris.csv"

# Load the dataset
dataset = pd.read_csv(url)


In [37]:
# Print the dataset
dataset

5.1  3.5  1.4  0.2     Iris-setosa
0    4.9  3.0  1.4  0.2     Iris-setosa
1    4.7  3.2  1.3  0.2     Iris-setosa
2    4.6  3.1  1.5  0.2     Iris-setosa
3    5.0  3.6  1.4  0.2     Iris-setosa
4    5.4  3.9  1.7  0.4     Iris-setosa
..   ...  ...  ...  ...             ...
144  6.7  3.0  5.2  2.3  Iris-virginica
145  6.3  2.5  5.0  1.9  Iris-virginica
146  6.5  3.0  5.2  2.0  Iris-virginica
147  6.2  3.4  5.4  2.3  Iris-virginica
148  5.9  3.0  5.1  1.8  Iris-virginica

[149 rows x 5 columns]

In [35]:
# Print the number of rows
Nrows = dataset.shape[0]
Nrows

149

In [36]:
# Print the number of columns
Ncols = dataset.shape[1]
Ncols

5

# Data types

# Activity: work with the iris dataset

1. Load the iris.csv file in your computer and understand the dataset


2. How many observations (rows) are in total?
- There are 149 rows in total.


3. How many variables (columns) are in total? What do they represent?
- There are 5 columns, they represent data types.

4. How many observations are for each type of flower?
- 50

5. What is the type of data for each variable?
- petal= float, sepal= float, stamen= float, stigma= float, type= objetc.

6. What are the units of each variable?
- cm for every one of them.
